libstatistics_collector/CMakeLists.txt

```diff
@@ -30,16 +30,23 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(rcl REQUIRED)
 find_package(rcpputils REQUIRED)
+find_package(rosidl_default_generators REQUIRED)
+find_package(std_msgs REQUIRED)
+
+rosidl_generate_interfaces(${PROJECT_NAME}
+  "test/msg/DummyMessage.msg"
+  DEPENDENCIES std_msgs
+)
 
 include_directories(include)
 
-add_library(${PROJECT_NAME} SHARED
+add_library(${PROJECT_NAME}_shared SHARED
```

Comment:
I think you can remove `_shared` by passing `LIBRARY_NAME`to `rosdl_generate_interfaces`

https://github.com/ros2/rosidl/blob/master/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake#L39